pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -140,6 +140,16 @@ func makeHookPod(hook *deployapi.LifecycleHook, deployment *kapi.ReplicationCont
 		restartPolicy = kapi.RestartPolicyOnFailure
 	}
 
+	// Transfer any requested volumes to the hook pod.
+	volumes := []kapi.Volume{}
+	for _, volume := range deployment.Spec.Template.Spec.Volumes {
+		for _, name := range exec.Volumes {
+			if volume.Name == name {
```

Comment:
@pmorie was talking about secret volumes so shouldn't this also include `&& volume.Secret != nil`?